iml-agent/src/action_plugins/ltuer.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+use std::path::{Path, PathBuf};
+use tokio::{
+    fs,
+    io::{AsyncWrite, AsyncWriteExt},
+};
+
+static CONFIGURATION_DIR: &str = "/etc/iml";
+
+async fn write_ltuer_conf<W>(
+    (mailbox_path, fs_name, cold_pool): (String, String, String),
+    writer: &mut W,
+) -> Result<(), ImlAgentError>
+where
+    W: AsyncWrite + Unpin,
+{
+    let contents = format!(
+        "mailbox={}\nfs_name={}\ncold_pool={}\n",
+        mailbox_path, fs_name, cold_pool
+    );
+
+    writer
+        .write(contents.as_bytes())
+        .await
+        .map(|_| ())
+        .map_err(|e| e.into())
+}
+
+async fn create_ltuer_conf_internal<P>(
+    (mailbox_path, fs_name, cold_pool): (String, String, String),
+    dir_path: P,
+) -> Result<(), ImlAgentError>
+where
+    P: AsRef<Path>,
+    PathBuf: From<P>,
+{
+    fs::create_dir_all(&dir_path).await?;
+
+    let mut path = PathBuf::from(dir_path);
+    path.push("ltuer.conf");
+
+    let mut file = fs::File::create(path).await?;
+
+    write_ltuer_conf((mailbox_path, fs_name, cold_pool), &mut file).await
+}
+
+pub async fn create_ltuer_conf(
+    (mailbox_path, fs_name, cold_pool): (String, String, String),
+) -> Result<(), ImlAgentError> {
+    create_ltuer_conf_internal((mailbox_path, fs_name, cold_pool), CONFIGURATION_DIR).await
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+
+    #[tokio::test]
+    async fn test_create_ltuer_conf() {
```

Comment:
nit: If you have this fn return  `Result<(), Box<dyn std::error::Error>>` then you can use the `?` operator instead of unwrapping

Reply:
I'm doing this on purpose, so that any unexpected error is fail-fast in test.

Reply:
It would still fail the same way, the question mark just returns the `Error` early.